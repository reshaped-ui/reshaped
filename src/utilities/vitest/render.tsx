import { render as vitestRender } from "vitest-browser-react";
import Reshaped from "components/Reshaped";
import "themes/reshaped/theme.css";

const Wrapper = (props: { children: React.ReactNode }) => {
	return <Reshaped>{props.children}</Reshaped>;
};

const render = (node: React.ReactNode) => {
	const screen = vitestRender(<div id="root">{node}</div>, {
		wrapper: Wrapper,
	});

	return { ...screen, container: document.getElementById("root")! };
};

export default render;
