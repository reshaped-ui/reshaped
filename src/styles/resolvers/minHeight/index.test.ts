import { expect, test, describe } from "vitest";
import getMinHeightStyles from "./index";

describe("Styles/MinHeight", () => {
	test("handles px value", () => {
		expect(getMinHeightStyles("50px")).toMatchSnapshot();
	});

	test("handles % value", () => {
		expect(getMinHeightStyles("50%")).toMatchSnapshot();
	});

	test("handles unit value", () => {
		expect(getMinHeightStyles(5)).toMatchSnapshot();
	});

	test("handles undefined value", () => {
		expect(getMinHeightStyles()).toMatchSnapshot();
	});

	test("handles responsive value", async () => {
		expect(getMinHeightStyles({ s: "50px", l: "50%" })).toMatchSnapshot();
	});
});
