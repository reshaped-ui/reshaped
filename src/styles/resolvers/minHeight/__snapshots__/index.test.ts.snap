// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Styles/MinHeight > handles % value 1`] = `
{
  "classNames": [
    "_root_940136",
    [
      "_--type-literal_940136",
    ],
  ],
  "variables": {
    "--rs-min-h-s": "50%",
  },
}
`;

exports[`Styles/MinHeight > handles px value 1`] = `
{
  "classNames": [
    "_root_940136",
    [
      "_--type-literal_940136",
    ],
  ],
  "variables": {
    "--rs-min-h-s": "50px",
  },
}
`;

exports[`Styles/MinHeight > handles responsive value 1`] = `
{
  "classNames": [
    "_root_940136",
    [
      "_--type-literal_940136",
      "_--type-literal--l_940136",
    ],
  ],
  "variables": {
    "--rs-min-h-l": "50%",
    "--rs-min-h-s": "50px",
  },
}
`;

exports[`Styles/MinHeight > handles undefined value 1`] = `{}`;

exports[`Styles/MinHeight > handles unit value 1`] = `
{
  "classNames": [
    "_root_940136",
    [
      "_--type-unit_940136",
    ],
  ],
  "variables": {
    "--rs-min-h-s": 5,
  },
}
`;
